python/amici/petab_import.py

```diff
@@ -477,8 +479,51 @@ def import_model(sbml_model: Union[str, 'libsbml.Model'],
         petab.add_global_parameter(sbml_model, par)
     # <EndWorkAround>
 
-    fixed_parameters = get_fixed_parameters(sbml_model=sbml_model,
-                                            condition_df=condition_df)
+    # TODO: to parameterize initial states or compartment sizes, we currently
+    #  need initial assignments. if they occur in the condition table, we
+    #  create a new parameter initial_${startOrCompartmentID}.
+    #  feels dirty and should be changed (see also #924)
+    # <BeginWorkAround>
+    initial_states = [col for col in condition_df
+                      if sbml_model.getSpecies(col) is not None]
+    initial_sizes = [col for col in condition_df
+                     if sbml_model.getCompartment(col) is not None]
+    fixed_parameters = []
+    if len(initial_states) or len(initial_sizes):
+        # add preequilibration indicator variable
+        # NOTE: would only be required if we actually have preequilibration
+        #  adding it anyways. can be optimized-out later
+        if sbml_model.getParameter(PREEQ_INDICATOR_ID) is not None:
+            raise AssertionError("Model already has a parameter with ID "
+                                 f"{PREEQ_INDICATOR_ID}. Cannot handle "
+                                 "species and compartments in condition table "
+                                 "then.")
+        indicator = sbml_model.createParameter()
+        indicator.setId(PREEQ_INDICATOR_ID)
+        indicator.setName(PREEQ_INDICATOR_ID)
+        # Can only reset parameters after preequilibration if they are fixed.
+        fixed_parameters.append(PREEQ_INDICATOR_ID)
+
+    for assignee_id in initial_sizes + initial_states:
+        init_par_id = f"initial_{assignee_id}"
+        if sbml_model.getElementBySId(init_par_id) is not None:
```

Comment:
are not {_preeq,_sim} added, not the sole init_par_id, so why check for that here? maybe misreading, just wanted to mention.

Reply:
Oops, that was from a previous version, where there was only one parameter for both and which did not support resetting concentrations after preeq.